resources/js/cart.js

```diff
@@ -0,0 +1,65 @@
+function postCartData(type, data) {
+    return fetch('/cart/' + type, {
+        method: 'POST',
+        headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(data)
+    }).then(res => res.json());
+}
+
+const cartController = (function() {
+    return {
+        updateItemCount: function(itemId, count, price, totalCount, totalPrice) {
+            count = count || 0;
+            if (count > 0) {
+                document.querySelectorAll('span[data-id="' + itemId + '"]')[0].innerHTML = count;
```

Comment:
You should search these elements with-in a scope where event has happened,  document.querySelectorAll does that for whole page.